cpp/models/abm/migration_rules.cpp

```diff
@@ -46,11 +46,13 @@ LocationType random_migration(const Person& person, TimePoint t, TimeSpan dt, co
     return current_loc;
 }
 
-LocationType go_to_school(const Person& person, TimePoint t, TimeSpan /*dt*/, const AbmMigrationParameters& params)
+LocationType go_to_school(const Person& person, TimePoint t, TimeSpan dt, const AbmMigrationParameters& params)
 {
     auto current_loc = person.get_location_id().type;
+    int seconds_of_day = 60*60*24;
+
     if (current_loc == LocationType::Home && t < params.get<LockdownDate>() && t.day_of_week() < 5 &&
-        t.hour_of_day() >= 8 && person.get_age() == AbmAgeGroup::Age5to14 && person.goes_to_school(t, params) &&
+        (t.seconds() - person.get_go_to_school_time(params, dt).seconds())%(seconds_of_day) == 0 && person.get_age() == AbmAgeGroup::Age5to14 && person.goes_to_school(t, params) &&
```

Comment:
I agree.
@dabele  Does this method(time_since_midnight) exist or I create it?

Reply:
does not exist yet but should be easy to implement, similar to the `hour_of_the_day()` method but not rounding to hours and returning a TimeSpan instead of int.

Reply:
Ok. This is done.